x/emissions/keeper/msgserver/msg_server_stake.go

```diff
@@ -144,12 +156,13 @@ func (ms msgServer) DelegateStake(ctx context.Context, msg *types.DelegateStakeR
 	if err := msg.Validate(); err != nil {
 		return nil, err
 	}
-	if msg.Amount.IsZero() {
-		return nil, types.ErrReceivedZeroAmount
-	}
 
-	if msg.Reputer == msg.Sender {
-		return nil, types.ErrCantSelfDelegate
+	topicExists, err := ms.k.TopicExists(ctx, msg.TopicId)
```

Comment:
I'm happy that you're being super paranoid checking and re-checking this.  We can optimize later.  Thank you